soroban-env-host/src/storage.rs

```diff
@@ -449,10 +449,15 @@ impl Storage {
         Ok(self.try_get_full_with_host(key, host, key_val)?.is_some())
     }
 
-    /// Extends `key` to live `extend_to` ledgers from now (not counting the
-    /// current ledger) if the current live_until_ledger for the entry is
+    /// Extends `key` to live `extend_by` ledgers from now (not counting the
```

Comment:
It's actually been correctly called `extend_to` (albeit it's a bit confusing). `extend_by` would mean that we take current TTL and extend it by a given number. Instead, we're not looking at the current TTL at all, just extend the TTL to be at least `extend_to` ledgers.

Reply:
This is an internal function so I think we should pick a name that best reflect what it actually does.  
All it does is adds the current sequence number by an amount, check some bounds, then set it as the new ttl value. I think the term that better interprets that amount is "extend by", i.e. new_live_until = sequence_number + extend_by.
We are not looking at TTL in this function as you mentioned, so I think making the input parameter the number the TTL is being extended to is pretty confusing.
Or do you think we can rename this function name to be more specific, e.g. `compute_new_live_until_and_set_ttl`.
I think this will help avoiding confusions such as mentioned in the tracking [issue](https://github.com/stellar/rs-soroban-env/issues/1329).

Reply:
But we do specifically extend the TTL, don't we? The equation you provided literally extends the TTL to be `extend_to/by` ledgers from now. I.e. the equation doesn't include the current live_until_ledger. This is also not quite internal, since you've updated the error message to include the new arg name.
We could try renaming this, but I'm not sure how does it address the issue, given that this function is indeed internal. The external interface has already been settled, for better or worse. We also don't really do two things here; TTL is just the *amount* of ledgers between now and live_until_ledger. I think if you have this definition in mind, this function would make more sense. The fact that we set it via setting a new `live_until_ledger` value is just an implementation detail; in an abstract sense this function extends TTL to be at least `extend_to` ledgers.

Reply:
Reverted the renaming change. Don't think this is worth arguing over right now. But I do feel the whole extension ttl, to include or not include current ledger is pretty confusing and footguny (as mentioned in the issue). If you think there are ways to improve it, I would be very glad.

Reply:
Yeah, I agree this is not ideal, though in the SDK things should be more consistent now, at least when SDK methods are being used. The way the ledger settings are interpreted is kind of confusing, but changing it would be a protocol change, and I'm not sure it's worth the hassle.